### PR TITLE
Development branch

### PR DESCRIPTION
**Pull request by scurest**
So now that @ksons is back (yaaay) I'm PRing my dev branch. Sorry for the big diff >_<

To give an overview of the changes

* the importer was split into multiple files (like you noticed :). The `__init__` file holds the operation class. It has a bunch of `get_X` methods that defer to `create_X` functions which cache their result. For example, when you call `get_mesh(id)` it creates a Blender mesh object for the glTF mesh with id `id` and caches it for the next call so Blender objects get reused.
* `buffer.py`  handles accessors and buffers. It is roughly feature-complete I think.
* `mesh.py` creates meshes. Each glTF `primitive` is turned into a Blender mesh with `from_pydata` and they're all joined into one mesh with `bmesh`.
* `node.py` handles nodes and scenes. It uses bones to position the objects. There's a long comment there about why. I'm pretty unhappy with the state of this but haven't been able to figure out the best way to map the glTF node idea onto Blender.
* `material.py` does textures and materials. Textures are loaded from memory by writing them to a temp file and loading that. The PBR pipeline is done with a group now so its a little more organized.
* tests should all pass; all the importer issues filed against this repo should be resolved (if they weren't already in #1)

The main missing glTF features are now

* node scaling (armatures don't seem to support this)
* animations (blocked on getting a good representation of the node forest)
* morph targets
* some material properties (occlusion textures, scaling, ...)
* camera properties (ortho, persp)

Um, yeah. If you have any questions or want any changes when you're reviewing I can make them.

### CODE REVIEW

File: io_scene_gltf/material.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+def create_material_from_properties(op, material, material_name):
+    pbr_metallic_roughness = material.get('pbrMetallicRoughness', {})
+
+    mat = bpy.data.materials.new(material_name)
+    op.materials[material_name] = mat
+    mat.use_nodes = True
+    tree = mat.node_tree
+    links = tree.links
+
+    for n in tree.nodes:
```
Comment (ksons) [inline io_scene_gltf/material.py]: Why is this necessary?

File: io_scene_gltf/node.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+COLLADA should have these problems too; check what it does? (I know it doesn't
+solve the bone positions issue.)
+
+For our purposes it would be really nice if Blender had an armature based on
+joints instead of bones.
+
+It would also be desirable to check that what we do gives the correst result ie.
+the transform Blender does equals the one glTF calls for, but neither Blender
+nor glTF have docs for exactly what that transform should be :-/
+
```
Comment (ksons) [inline io_scene_gltf/node.py]: I don't get the last paragraph. Can you explain?

**Comment by ksons** [comment]
I did not fully understand your armature approach (I need to look deeper into it) and would indeed prefer a more straightforward mapping from glTF entities to Blender entities. But well, if it cannot be mapped easily, yours seem to be the current solution and let's see if we have a 💡 moment to simplify.

Thanks again for the great progress!

File: io_scene_gltf/material.py (hunk 1 above)
Comment (scurest) [reply io_scene_gltf/material.py]: It just gives us a clean slate to work on. A new material comes with some nodes already in it (a material and output node).

File: io_scene_gltf/node.py (hunk 2 above)
Comment (scurest) [reply io_scene_gltf/node.py]: About scenes? If the glTF has nodes like this

       0       3
      / \     / \
     1   2   4   5
         |       |
       mesh1   mesh2

and there's a scene with root 0 and another with root 3, then we turn that into:

1. a Blender armature object A for all the nodes
2. two meshes, M1 for mesh1 and M2 for mesh 2, which we attach to A
3. for the first scene, we link in A and M1
4. for the second scene, we link in A and M2

Basically, for each scene, we link in A and then any object which is a descendant of one of the scene's roots.

In an older version I tried to partition it up so that there would be different armatures for the different scenes (eg. the first scene wouldn't contain node 3, 4, or 5) but the scenes can overlap and mesh1 could be skinned with eg. node 3, so I gave that up.

The glTF spec doesn't say much about what scenes actually do. The only prescriptive thing I could find was that the default scene should be "displayed at load time". Other than that a scene just marks some of the nodes in the forest as "roots".

I was also kind of considering just dropping scene handling too.
